src/psyclone/gocean1p0.py

```diff
@@ -880,6 +881,222 @@ def backward_dependence(self):
+        indicates a particular type of stencil or in the form
+        stencil(xxx,yyy,zzz) which explicitly specifies a stencil
+        shape where xxx, yyy and zzz are triplets of integers
+        indicating whether there is a stencil access in a particular
+        direction and the depth of that access. For example:
+
+        stencil(010,  !   N
+                212,  !  W E
+                010)  !   S
+
+        indicates that there is a stencil access of depth 1 in the
+        "North" and "South" directions and and stencil access of depth
+        1 and 2 in the "East" and "West" directions. The value at the
+        centre of the stencil will not be used by PSyclone but can be
+        0 or 1 and indicates whether the local field value is
+        accessed.
+
+        The convention is for the associated arrays to be
+        2-dimensional. If we denote the first dimension as "i" and the
+        second dimension as "j" then the following directions are
+        assumed:
+
+        j
+        ^
+        |
+        |
+        ---->i
+
+        For example a stencil access like:
+
+        a(i,j) + a(i+1,j) + a(i,j-1)
+
+        would be stored as:
+
+        stencil(000,
+                011,
+                010)
+
+        :param stencil_info: contains the appropriate part of the parser AST
+        :type stencil_info: :py:class:`psyclone.expression.FunctionVar`
+        :param string kernel_name: the name of the kernel from where
+        this stencil information came from.
+        :raises ParseError: if the supplied stencil information is invalid
+
+        '''
+        self._initialised = True
+        if isinstance(stencil_info, expr.FunctionVar):
+            # Get the name
+            name = stencil_info.name.lower()
+
+            if stencil_info.args:
+                # The stencil info is of the form 'name(a,b,...), so
+                # the name should be 'stencil' and there should be 3
+                # arguments'
+                self._has_stencil = True
+                args = stencil_info.args
+                if name != "stencil":
+                    raise ParseError(
+                        "Meta-data error in kernel '{0}': 3rd descriptor "
+                        "(stencil) of field argument is '{1}' but must be "
+                        "'stencil(...)".format(kernel_name, name))
+                if len(args) != 3:
+                    raise ParseError(
+                        "Meta-data error in kernel '{0}': 3rd descriptor "
+                        "(stencil) of field argument with format "
+                        "'stencil(...)', has {1} arguments but should have "
+                        "3".format(kernel_name, len(args)))
+                # Each of the 3 args should be of length 3 and each
+                # character should be a digit from 0-9
+                for arg_idx in range(3):
+                    arg = args[arg_idx]
+                    if not isinstance(arg, str):
```

Comment:
You check that `arg` is a `str` but then the error message says that `arg` should be a number...

Reply:
You could use a regex to check that you have three digits perhaps?

Reply:
OK, thanks for clarifying.